src/components/shared/FilterNavigatorBar.css

```diff
@@ -125,25 +132,30 @@
 
 /* Animation */
 
+.filterNavigatorBarUncommittedTransition-exit {
+  /* Because of the underlying transition library, this element is still here
+   * while the new "committed" element is created, which pushes it further
+   * right. By using display: none here, we prevent this bad effect. */
+  display: none;
+}
+
 .filterNavigatorBarTransition-enter {
-  z-index: 0;
-  opacity: 0.1;
-  transform: translateX(-100%);
+  color: inherit;
+  opacity: 0.65;
 }
 
 .filterNavigatorBarTransition-enter.filterNavigatorBarTransition-enter-active {
+  color: var(--internal-selected-color);
   opacity: 1;
```

Comment:
I tried removing these or the ones right above, but there was always some case that wouldn't work well.